GraphAlgorithms/Utils/ModularProduct.cs

```diff
@@ -43,18 +26,27 @@ private static Graph Calculate(Graph graph1, Graph graph2, bool includeWeights)
                     int u2 = j / n;
                     int v2 = j % n;
 
-                    if (u1 != u2 && v1 != v2
-                        && graph1.AdjacencyMatrix[u1, u2] == graph2.AdjacencyMatrix[v1, v2]
-                        && graph1.AdjacencyMatrix[u2, u1] == graph2.AdjacencyMatrix[v2, v1])
+                    if (u1 == u2 || v1 == v2)
+                        continue;
+                    
+                    if (graph1.AdjacencyMatrix[u1, u2] == 0)
```

Comment:
can all of these conditions be written as:
```
if (graph1.AdjacencyMatrix[u1, u2] == 0 && graph2.AdjacencyMatrix[v1, v2] == 0)
    product.AdjacencyMatrix[i, j] = 1;

product.AdjacencyMatrix[i, j] = Math.Min(graph1.AdjacencyMatrix[u1, u2], graph2.AdjacencyMatrix[v1, v2]);

if (graph1.AdjacencyMatrix[u2, u1] == 0 && graph2.AdjacencyMatrix[v2, v1] == 0)
    product.AdjacencyMatrix[j, i] = 1;

product.AdjacencyMatrix[j, i] = Math.Min(graph1.AdjacencyMatrix[u2, u1], graph2.AdjacencyMatrix[v2, v1]);
```

Reply:
I'm asking to check if I understand correctly, because maybe this code is equivalent to what I wrote above (it passes the tests anyway). Another thing, It's a little odd that we have three assignments to [i,j] and only one to [j,i] , maybe it should be like that but it's non-symmetric, that;s why im asking

Reply:
the [i,j] was a mistake and it makes me worried that the tests passed

and yes you can rewrite it like that but with an extra 2 *